noir/noir-repo/acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -82,74 +91,101 @@ pub(crate) fn evaluate_binary_int_op<F: AcirField>(
         BinaryIntOp::Div => {
-            if rhs.is_zero() {
-                BigUint::zero()
+            if rhs == 0 {
+                0
             } else {
                 lhs / rhs
             }
         }
-        // Perform a == operation, returning 0 or 1
-        BinaryIntOp::Equals => {
-            if lhs == rhs {
-                BigUint::one()
+        BinaryIntOp::Equals => (lhs == rhs) as u128,
+        BinaryIntOp::LessThan => (lhs < rhs) as u128,
+        BinaryIntOp::LessThanEquals => (lhs <= rhs) as u128,
+        BinaryIntOp::And => lhs & rhs,
+        BinaryIntOp::Or => lhs | rhs,
+        BinaryIntOp::Xor => lhs ^ rhs,
+        BinaryIntOp::Shl => {
+            if rhs >= 128 {
+                0
             } else {
-                BigUint::zero()
+                lhs.wrapping_shl(rhs as u32)
             }
         }
-        // Perform a < operation, returning 0 or 1
-        BinaryIntOp::LessThan => {
-            if lhs < rhs {
-                BigUint::one()
+        BinaryIntOp::Shr => {
+            if rhs >= 128 {
+                0
             } else {
-                BigUint::zero()
+                lhs.wrapping_shr(rhs as u32)
             }
         }
-        // Perform a <= operation, returning 0 or 1
-        BinaryIntOp::LessThanEquals => {
-            if lhs <= rhs {
-                BigUint::one()
+    }
+}
+
+fn evaluate_binary_int_op_generic(
+    op: &BinaryIntOp,
+    lhs: u128,
+    rhs: u128,
+    bit_size: IntegerBitSize,
+) -> u128 {
+    let bit_size: u32 = bit_size.into();
+    let bit_modulo = 1 << bit_size;
+    match op {
+        // Perform addition, subtraction, and multiplication, applying a modulo operation to keep the result within the bit size.
+        BinaryIntOp::Add => (lhs + rhs) % bit_modulo,
+        BinaryIntOp::Sub => (bit_modulo + lhs - rhs) % bit_modulo,
+        BinaryIntOp::Mul => (lhs * rhs) % bit_modulo,
+        // Perform unsigned division using the modulo operation on a and b.
+        BinaryIntOp::Div => {
+            if rhs == 0 {
+                0
             } else {
-                BigUint::zero()
+                lhs / rhs
             }
         }
+        // Perform a == operation, returning 0 or 1
+        BinaryIntOp::Equals => (lhs == rhs) as u128,
+        // Perform a < operation, returning 0 or 1
+        BinaryIntOp::LessThan => (lhs < rhs) as u128,
+        // Perform a <= operation, returning 0 or 1
+        BinaryIntOp::LessThanEquals => (lhs <= rhs) as u128,
         // Perform bitwise AND, OR, XOR, left shift, and right shift operations, applying a modulo operation to keep the result within the bit size.
         BinaryIntOp::And => lhs & rhs,
         BinaryIntOp::Or => lhs | rhs,
         BinaryIntOp::Xor => lhs ^ rhs,
         BinaryIntOp::Shl => {
-            if bit_size > 128 {
-                return Err(BrilligArithmeticError::InvalidShift { op_bit_size: bit_size });
+            if rhs >= (bit_size as u128) {
```

Comment:
here also I think the behaviour is different from Noir.
We should error if rhs >= bit_size

Reply:
Cool, will change that!